gtsam/linear/GaussianFactorGraph-impl.h

```diff
@@ -0,0 +1,100 @@
+/* ----------------------------------------------------------------------------
```

Comment:
Why templated?? If currently we only use double, undo the template and just stick it into gfg.cpp, please. We can always template *when* we need it.

Reply:
Yes, I will fix this.

The trouble was that SparseEigen needed to call this function with template parameter `vector<Eigen::Triplet>` which is defined in `<Eigen/Sparse>`, so defining it in `gfg.cpp` would require either including `<Eigen/Sparse>` or forward declaring something that the caller must include.  I thought for a long time about how avoid templating and previously, I couldn't think of a way to avoid it.

*However*, once I made the "efficiency improvements" then I found 2 ways around it:
1) Just define our own version of Triplet that's more efficient than boost::tuple<size_t, size_t, double>
2) Use boost::tuple and be slightly less efficient (as per the table, about 12% slower).

I will do the former.